File: ANewFolder/MyCards-master (2)/app/src/main/java/edu/univdhaka/cse/cse2216/mycards/java/activities/CardDetailsActivity.java
package edu.univdhaka.cse.cse2216.mycards.java.activities;

import android.annotation.SuppressLint;
import android.app.Activity;
import android.app.AlertDialog;
import android.content.DialogInterface;
import android.content.Intent;
import android.os.Bundle;
import android.util.Log;
import android.view.Menu;
import android.view.MenuItem;
import android.widget.TextView;
import android.widget.Toast;

import edu.univdhaka.cse.cse2216.mycards.R;
import edu.univdhaka.cse.cse2216.mycards.java.domain.Card;
import edu.univdhaka.cse.cse2216.mycards.java.service.CardService;

public class CardDetailsActivity extends Activity {

    private Card card;

    private TextView cardTypeLabel;
    private TextView bankNameLabel;
    private TextView cardNumberLabel;
    private TextView cardholderNameLabel;
    private TextView expiryDateLabel;
    private TextView cvvLabel;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        setTitle(R.string.card_details);

        setContentView(R.layout.activity_card_details);

        bindWidgets();

        if (getIntent().getExtras() != null) {
            Object card = getIntent().getExtras().get("card");
            if (card != null) {
                this.card = (Card) card;
            }
        }

        bindValuesToLabels();
    }

    private void bindWidgets() {
        cardTypeLabel = findViewById(R.id.card_type_label);
        bankNameLabel = findViewById(R.id.bank_name_label);
        cardNumberLabel = findViewById(R.id.card_number_label);
        cardholderNameLabel = findViewById(R.id.cardholder_name_label);
        expiryDateLabel = findViewById(R.id.expiry_date_label);
        cvvLabel = findViewById(R.id.cvv_label);
    }

    @SuppressLint("SetTextI18n")
    private void bindValuesToLabels() {
        cardTypeLabel.setText(card.getType());
        bankNameLabel.setText(card.getBankName());
        cardNumberLabel.setText(card.getNumber());
        cardholderNameLabel.setText(card.getCardholderName());
        expiryDateLabel.setText(card.getExpiryDate());
        cvvLabel.setText(Integer.toString(card.getCvv()));
    }

    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        getMenuInflater().inflate(R.menu.menu_card_details, menu);
        return super.onCreateOptionsMenu(menu);
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        switch (item.getItemId()) {
            case R.id.action_edit:
                onEdit();
                return true;
            case R.id.action_delete:
                onDelete();
                return true;
        }

        return super.onOptionsItemSelected(item);
    }

    private void onEdit() {
        startActivity(new Intent(this, AddEditCardActivity.class).putExtra("card", card));
    }

    private void onDelete() {
        new AlertDialog.Builder(this)
                .setTitle(R.string.delete_card_confirmation_title)
                .setMessage(R.string.delete_card_confirmation_message)
                .setPositiveButton(R.string.delete,
                        new DialogInterface.OnClickListener() {
                            @Override
                            public void onClick(DialogInterface dialog, int which) {
                                deleteCard();
                            }
                        })
                .setNegativeButton(R.string.cancel, null)
                .create()
                .show();
    }

    @SuppressWarnings("DebugLoggingFunction")
    private void deleteCard() {
        Log.d(getString(R.string.app_name), "Delete card: " + card.toString());

        CardService cardService = new CardService(this);
        cardService.deleteCard(card);

        Toast.makeText(this, R.string.card_delete_success, Toast.LENGTH_SHORT).show();

        finish();
    }
}

File: ANewFolder/MyCards-master (2)/app/src/main/java/edu/univdhaka/cse/cse2216/mycards/kotlin/activities/CardDetailsActivity.kt
package edu.univdhaka.cse.cse2216.mycards.kotlin.activities

import android.app.Activity
import android.app.AlertDialog
import android.content.Intent
import android.os.Bundle
import android.util.Log
import android.view.Menu
import android.view.MenuItem
import android.widget.TextView
import android.widget.Toast
import edu.univdhaka.cse.cse2216.mycards.R
import edu.univdhaka.cse.cse2216.mycards.kotlin.service.CardService
import edu.univdhaka.cse.cse2216.mycards.kotlin.domain.Card

class CardDetailsActivity: Activity() {

    private lateinit var card: Card

    private lateinit var cardTypeLabel: TextView
    private lateinit var bankNameLabel: TextView
    private lateinit var cardNumberLabel: TextView
    private lateinit var cardholderNameLabel: TextView
    private lateinit var expiryDateLabel: TextView
    private lateinit var cvvLabel: TextView

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        setTitle(R.string.card_details)

        setContentView(R.layout.activity_card_details)

        bindWidgets()

        card = intent.extras?.get("card") as Card

        bindValuesToLabels()
    }

    private fun bindWidgets() {
        cardTypeLabel = findViewById(R.id.card_type_label)
        bankNameLabel = findViewById(R.id.bank_name_label)
        cardNumberLabel = findViewById(R.id.card_number_label)
        cardholderNameLabel = findViewById(R.id.cardholder_name_label)
        expiryDateLabel = findViewById(R.id.expiry_date_label)
        cvvLabel = findViewById(R.id.cvv_label)
    }

    private fun bindValuesToLabels() {
        cardTypeLabel.text = card.type
        bankNameLabel.text = card.bankName
        cardNumberLabel.text = card.number
        cardholderNameLabel.text = card.cardholderName
        expiryDateLabel.text = card.expiryDate
        cvvLabel.text = card.cvv.toString()
    }

    override fun onCreateOptionsMenu(menu: Menu): Boolean {
        menuInflater.inflate(R.menu.menu_card_details, menu)
        return true
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        return when (item.itemId) {
            R.id.action_edit -> {
                onEdit()
                true
            }
            R.id.action_delete -> {
                onDelete()
                true
            }
            else -> super.onOptionsItemSelected(item)
        }
    }

    private fun onEdit() {
        startActivity(Intent(this, AddEditCardActivity::class.java).putExtra("card", card))
    }

    private fun onDelete() {
        AlertDialog.Builder(this)
                .setTitle(R.string.delete_card_confirmation_title)
                .setMessage(R.string.delete_card_confirmation_message)
                .setPositiveButton(R.string.delete){ _, _ -> deleteCard() }
                .setNegativeButton(R.string.cancel){ _, _ -> }
                .create()
                .show()
    }

    private fun deleteCard() {
        Log.d(getString(R.string.app_name), "Delete card: " + card.toString())

        val cardService = CardService(this)
        cardService.deleteCard(card)

        Toast.makeText(this, R.string.card_delete_success, Toast.LENGTH_SHORT).show()

        finish()
    }
}
